opencaesar/oml-vision
# Feat/29-property-sheet

## Checklist before submitting a merge request


## Description of contribution

Opens a property sheet when selecting an element in OML Vision

closes https://github.com/opencaesar/oml-vision/issues/29
 
## Testing performed




## How to Test Expected functionality changes

OML Model:
1. You can use this OML model for testing https://github.com/pogi7/kepler16b-example
2. The `propertyLayouts.json` file is already present, but verify that following JSON is defined in the `src/vision/viewpoints/properties/propertyLayouts.json` file.

```json
{
  "pages": [
    {
      "id": "general",
      "label": "General",
      "sparqlQuery": "general-properties.sparql",
      "icon": "IconSettings",
      "groups": [
        {
          "id": "instance",
          "label": "Instance",
          "controls": [
            {
              "type": "text",
              "id": "id",
              "label": "Label",
              "helpExpression": "A label can provide a human-readable id for model objects that have assigned ids that might be less readable"
            },
            {
              "type": "text",
              "id": "anyIri",
              "label": "IRI",
              "readOnly": true,
              "helpExpression": "This is the model-assigned element identifier. Every model element has this unique identifier"
            },
            {
              "type": "text",
              "id": "name",
              "label": "Name",
              "helpExpression": "Lists zero or more model types that this instance specializes. Such supertypes carry inherited properties or relations in the model."
            }
          ]
        }
      ]
    }
  ]
}
```

3. Verify that if you open the `Objectives` table and click the first row, this property sheet pops up (BUG: you may need to click on the row multiple times for the sheet to render properly)
![image](https://github.com/opencaesar/oml-vision/assets/108915844/269093ff-c2e6-4484-8881-7172b768e5ba)
4. Verify that if you open the `Missions` tree and click the first row, this property sheet pops up (BUG: you may need to click on the row multiple times for the sheet to render properly)
![image](https://github.com/opencaesar/oml-vision/assets/108915844/cbc05823-cbd6-4ef4-9a7c-fc6395dbb3bc)
5. Verify that if you open the `Components` diagram and click the `C.02.09` node, this property sheet pops up (BUG: you may need to click on the node multiple times for the sheet to render properly)
![image](https://github.com/opencaesar/oml-vision/assets/108915844/4313edb6-d7ad-408e-96b6-73cc3b71cd0a)


## Additional context
